src/treemodel/range.js

```diff
@@ -261,32 +252,51 @@ export default class Range {
 	}
 
 	/**
-	 * Returns an iterator that iterates over all {@link core.treeModel.Node nodes} that are in this range and returns them.
+	 * Returns an iterator that iterates over all {@link core.treeModel.Item items} that are in this range and returns
+	 * them together together with additional information like length or {@link core.treeModel.Position positions},
+	 * grouped as {@link core.treeModel.TreeWalkerValue}.
+	 *
 	 * A node is in the range when it is after a {@link core.treeModel.Position position} contained in this range.
-	 * In other words, this iterates over all text characters that are inside the range and all the {@link core.treeModel.Element}s
-	 * we enter into when iterating over this range.
+	 * In other words, this iterates over all text characters that are inside the range and all the
+	 * {@link core.treeModel.Element}s we enter into when iterating over this range.
+	 *
+	 * **Note:** this method will not return a parent node of start position. This is in contrary to
+	 * {@link core.treeModel.TreeWalker} which will return that node with `'ELEMENT_END'` type. This method also
+	 * returns each {@link core.treeModel.Element} once, while simply used {@link core.treeModel.TreeWalker} might
+	 * return it twice: for `'ELEMENT_START'` and `'ELEMENT_END'`.
 	 *
-	 * **Note:** this method will not return a parent node of start position. This is in contrary to {@link core.treeModel.TreeWalker}
-	 * which will return that node with `'ELEMENT_END'` type. This method also returns each {@link core.treeModel.Element} once,
-	 * while simply used {@link core.treeModel.TreeWalker} might return it twice: for `'ELEMENT_START'` and `'ELEMENT_END'`.
+	 * **Node:*** because this method does not return {@link core.treeModel.TreeWalkerValue values} with the type of
+	 * `'ELEMENT_END'`, you can use {@link core.treeModel.TreeWalkerValue.previousPosition} as a position before the
+	 * item.
 	 *
 	 * @see {treeModel.TreeWalker}
 	 * @param {Boolean} [mergeCharacters] Flag indicating whether all consecutive characters with the same attributes
-	 * should be returned as one {@link core.treeModel.TextFragment} (`true`) or one by one as multiple {@link core.treeModel.CharacterProxy}
-	 * (`false`) objects. Defaults to `false`.
-	 * @returns {Iterable.<core.treeModel.Node|treeModel.TextFragment>}
+	 * should be returned as one {@link core.treeModel.TextFragment} (`true`) or one by one as multiple
+	 * {@link core.treeModel.CharacterProxy} (`false`) objects. Defaults to `false`.
+	 * @returns {Iterable.<core.treeModel.TreeWalkerValue>}
 	 */
-	*getAllNodes( mergeCharacters ) {
-		let it = new TreeWalker( { boundaries: this, mergeCharacters: mergeCharacters } );
-		let step;
+	*getValues( mergeCharacters ) {
```

Comment:
I removed default iterator, because we do not know if one what to `mergeCharacters` or not and if one want to return `ELEMENT_END` or not. I agree it was useless with the form it was. For me we could make this method default iterator with the `mergeCharacters`, so it will skip `ELEMENT_END` and merge characters. It is the most convenient for me, but I do not know if it is what people expect.